src/jdk.compiler/share/classes/com/sun/tools/javac/code/Lint.java

```diff
@@ -460,9 +448,8 @@ private void suppress(LintCategory lc) {
 
         public void visitConstant(Attribute.Constant value) {
             if (value.type.tsym == syms.stringType.tsym) {
-                LintCategory lc = LintCategory.get((String) (value.value));
-                if (lc != null)
-                    suppress(lc);
+                LintCategory.get((String)value.value)
```

Comment:
I think use of `Optional` makes sense here too, so bonus point for the extra consolidation!